internal/gcs/guestcaps.go

```diff
@@ -0,0 +1,61 @@
+//go:build windows
+
+package gcs
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	"github.com/Microsoft/hcsshim/internal/hcs/schema1"
+)
+
+// GuestDefinedCapabilities is an interface for different guest defined capabilities.
```

Comment:
Is there an advantage of having an interface instead of having a single type that contains fields for capabilities of both LCOW & WCOW? When the struct of that type is created for say LCOW, all the WCOW specific capabilities can just be `false` or `nil`. That way the caller won't need to worry about casting the interface into appropriate struct.

Reply:
The inspiration behind this change was that eventually we may be in a place where we want to add a lot of features to LCOW that break the contract between the shim and the gcs. But today we use the schema1 definition for GuestDefinedCapabilities and I think that makes it unclear what features are actually supported by both and I think that type could eventually be bogged down in fields that are only supported by LCOW. 

My goal overall was just to make it easier for people to add new guest defined capabilities for LCOW, but if we think this is not the right solution, I'm happy to investigate other options :)